pandas/core/frame.py

```diff
@@ -6932,7 +6932,7 @@ def value_counts(
 
         Parameters
         ----------
-        subset : list-like, optional
+        subset : mapping, function, label, list of labels, optional
             Columns to use when counting unique combinations.
```

Comment:
For mapping and function, I think we need to indicate how they are used. Saying "Columns to use when counting unique combinations" isn't correct in this case, right?

Reply:
Do you think the following would work?

 Columns to use when counting unique combinations. If ```subset``` is a function, it's called on each value of the object's index. If a mapping is passed, the Series or dict VALUES will be used to determine the columns or groups (the Series' values are first aligned; see ```align()``` method).

This was derived from the [by](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.groupby.html) definition in the groupby docs page.

Reply:
If `subset` does not support `len` (in particular, functions), then it appears to me the current implementation will raise. I would also recommend _not_ supporting functions. It may give our users whiplash to have a method that does value_counts on the columns all of a sudden doing it on the index :laughing: !

I find the mapping argument case quite odd. I don't think I'd expect `df.value_counts(subset={'a': 1, 'b': 2})` to work, but if I must choose a behavior, then my first guess would be that it behaves the same as `df.value_counts(subset=list({'a': 1, 'b': 2}))` which would be using subset on the keys rather than values. I'm not strongly opposed to including mapping though, and if we do I think what you have looks good.

For supporting single-label arguments, the implementation where we check `len(subset)` needs to be improved.

Reply:
@rhshadrach I agree with you and will make it just say `subset : label, list of labels, optional` and then I think the saying "Column(s) to use when counting unique combinations" would be appropriate.